test/datatype/reduce_local.c

```diff
@@ -44,19 +44,12 @@ static op_name_t array_of_ops [] = {
     { "replace", "MPI_REPLACE", MPI_REPLACE },
     { NULL, "MPI_OP_NULL", MPI_OP_NULL }
 };
-static int do_ops[12] = { -1, };  /* index of the ops to do. Size +1 larger than the array_of_ops */
+static int global_do_ops[12] = { -1, };  /* index of the ops to do. Size +1 larger than the array_of_ops */
 static int verbose = 0;
 static int total_errors = 0;
 
-#define max(a,b) \
-   ({ __typeof__ (a) _a = (a); \
-       __typeof__ (b) _b = (b); \
-     _a > _b ? _a : _b; })
-
-#define min(a,b) \
-   ({ __typeof__ (a) _a = (a); \
-       __typeof__ (b) _b = (b); \
-     _a < _b ? _a : _b; })
+#define max(a,b) ((a) > (b) ? (a) : (b))
```

Comment:
Hey @bosilca -- can you comment on this?  I replaced the prior `min` and `max` macros with the canonical, simpler versions, but I don't know if there was something subtle happening in the old macros (e.g., with making new variable instances via `__typeof__`).  The warning that the compiler was emitting was that ISO C forbids making new variable instances inside expressions.

Do we need something better than the canonical `min`/`max` that I introduced here?

Reply:
@bwbarrett pointed out to me that the `opal/util/minmax.h` macros are probably a good choice here.

Reply:
Definitively better than the originally proposed replacement.